librosa/util/utils.py

```diff
@@ -35,80 +35,168 @@
 
     >>> y, sr = librosa.load(librosa.util.example_audio_file())
-    >>> librosa.util.frame(y, frame_length=2048, hop_length=64)
-    array([[ -9.216e-06,   7.710e-06, ...,  -2.117e-06,  -4.362e-07],
-           [  2.518e-06,  -6.294e-06, ...,  -1.775e-05,  -6.365e-06],
+    >>> frames = librosa.util.frame(y, frame_length=2048, hop_length=64)
+    >>> frames
+    array([[ 0.000e+00,  0.000e+00, ..., -2.448e-06, -6.789e-07],
+           [ 0.000e+00,  0.000e+00, ..., -1.399e-05,  1.004e-06],
            ...,
-           [ -7.429e-04,   5.173e-03, ...,   1.105e-05,  -5.074e-06],
-           [  2.169e-03,   4.867e-03, ...,   3.666e-06,  -5.571e-06]], dtype=float32)
+           [-7.352e-04,  5.162e-03, ...,  0.000e+00,  0.000e+00],
+           [ 2.168e-03,  4.870e-03, ...,  0.000e+00,  0.000e+00]],
+          dtype=float32)
+    >>> y.shape
+    (1355168,)
+    >>> frames.shape
+    (2048, 21143)
+
+    Or frame along the first axis instead of the last:
+
+    >>> frames = librosa.util.frame(y, frame_length=2048, hop_length=64, axis=0)
+    >>> frames.shape
+    (21143, 2048)
 
+    Frame a stereo signal:
+
+    >>> y, sr = librosa.load(librosa.util.example_audio_file(), mono=False)
+    >>> y.shape
+    (2, 1355168)
+    >>> frames = librosa.util.frame(y, frame_length=2048, hop_length=64)
+    (2, 2048, 21143)
+
+    Carve an STFT into fixed-length patches of 32 frames with no overlap
+
+    >>> y, sr = librosa.load(librosa.util.example_audio_file())
+    >>> S = np.abs(librosa.stft(y))
+    >>> S.shape
+    (1025, 2647)
+    >>> S_patch = librosa.util.frame(S, frame_length=32, hop_length=32)
+    >>> S_patch.shape
+    (1025, 32, 82)
+    >>> # The first patch is equivalent to the first 32 frames of S
+    >>> np.allclose(S_patch[:, :, 0], S[:, :32])
+    True
     '''
 
-    if not isinstance(y, np.ndarray):
+    if not isinstance(x, np.ndarray):
         raise ParameterError('Input must be of type numpy.ndarray, '
-                             'given type(y)={}'.format(type(y)))
+                             'given type(x)={}'.format(type(x)))
 
-    if y.ndim != 1:
-        raise ParameterError('Input must be one-dimensional, '
-                             'given y.ndim={}'.format(y.ndim))
-
-    if len(y) < frame_length:
-        raise ParameterError('Buffer is too short (n={:d})'
-                             ' for frame_length={:d}'.format(len(y), frame_length))
+    if x.shape[axis] < frame_length:
+        raise ParameterError('Input is too short (n={:d})'
+                             ' for frame_length={:d}'.format(x.shape[axis], frame_length))
 
     if hop_length < 1:
         raise ParameterError('Invalid hop_length: {:d}'.format(hop_length))
 
-    if not y.flags['C_CONTIGUOUS']:
-        raise ParameterError('Input buffer must be contiguous.')
+    n_frames = 1 + (x.shape[axis] - frame_length) // hop_length
+    strides = np.asarray(x.strides)
+
+    if axis == -1:
```

Comment:
what about `axis=x.ndims`? would that be a synonym of `axis=-1`?

Reply:
technically yes, but -1 is a more standard convention for this sort of thing (see eg np.stack)

Reply:
you're right. upon reflection, it would be a bad idea to support more cases than `0` and `-1`.